github/projects.go

```diff
@@ -65,6 +65,8 @@ type ProjectOptions struct {
 	Name string `json:"name,omitempty"`
 	// The body of the project. (Optional.)
 	Body string `json:"body,omitempty"`
+	// State of the project. Either open or closed
```

Comment:
Add a missing period to the end of the last sentence.

Also, let's figure if this is optional or required, and document that too, just like the two fields above.

Reply:
From looking over https://developer.github.com/v3/projects/, it seems that this field is optional and only applicable to the `UpdateProject` endpoint. Let's write that:

```Go
// State of the project. Either "open" or "closed". (Only for update; optional.)
State string `json:"state,omitempty"`
```

I wonder if it's okay that we're reusing the same `ProjectOptions` for updating and creating projects, given that creating projects doesn't have `State` field...

Reply:
@shurcooL The `state` field is invalid for Project Creation.

It is optional for Project Updates though.

In this case I think it'd warrant creating its own `ProjectUpdateOptions` struct since it's not valid for Project Creation to have the `state` field.

wdyt?

Reply:
Perhaps we should organize the fields like this:

```Go
Name string `json:"name,omitempty"` // The name of the project. (Required for creation; optional for update.)
Body string `json:"body,omitempty"` // The body of the project. (Optional.)

// The following fields are only applicable for update.
// They should be left with zero values for creation.
State string `json:"state,omitempty"` // State of the project. Either "open" or "closed". (Optional.)
```

Reply:
> In this case I think it'd warrant creating its own `ProjectUpdateOptions` struct since it's not valid for Project Creation to have the `state` field.

Yes, it's possible for us to do that too. It would be a breaking API change.

I'm not quite sure which of the two approaches would be better, need to think about it.

Reply:
We do have a history of reusing same struct for creating and updating certain resources (e.g., [issues](https://godoc.org/github.com/google/go-github/github#IssueRequest), [comments](https://godoc.org/github.com/google/go-github/github#IssueComment)).

I think let's go with my suggested approach of documenting the separate update-only fields for now, and reusing the same struct. They don't deviate very strongly at this point. Should that change in the future, we can make the breaking API change _then_.